eclair-core/src/main/scala/fr/acinq/eclair/blockchain/bitcoind/rpc/ExtendedBitcoinClient.scala

```diff
@@ -149,26 +150,36 @@ class ExtendedBitcoinClient(val rpcClient: BitcoinJsonRPCClient) {
 
   def validate(c: ChannelAnnouncement)(implicit ec: ExecutionContext): Future[ValidateResult] = {
     val TxCoordinates(blockHeight, txIndex, outputIndex) = coordinates(c.shortChannelId)
-
-    for {
-      blockHash: String <- rpcClient.invoke("getblockhash", blockHeight).map(_.extractOrElse[String](ByteVector32.Zeroes.toHex))
-      txid: String <- rpcClient.invoke("getblock", blockHash).map {
-        case json => Try {
-          val JArray(txs) = json \ "tx"
-          txs(txIndex).extract[String]
-        } getOrElse ByteVector32.Zeroes.toHex
-      }
-      tx <- getRawTransaction(txid)
-      unspent <- isTransactionOutputSpendable(txid, outputIndex, includeMempool = true)
-      fundingTxStatus <- if (unspent) {
-        Future.successful(UtxoStatus.Unspent)
-      } else {
-        // if this returns true, it means that the spending tx is *not* in the blockchain
-        isTransactionOutputSpendable(txid, outputIndex, includeMempool = false).map {
-          case res => UtxoStatus.Spent(spendingTxConfirmed = !res)
+      val span = Kamon.spanBuilder("validate-bitcoin-client").start()
+      for {
+        _ <- Future.successful(0)
+        span0 = Kamon.spanBuilder("getblockhash").start()
+        blockHash: String <- rpcClient.invoke("getblockhash", blockHeight).map(_.extractOrElse[String](ByteVector32.Zeroes.toHex))
+        _ = span0.finish()
+        span1 = Kamon.spanBuilder("getblock").start()
+        txid: String <- rpcClient.invoke("getblock", blockHash).map {
+          case json => Try {
+            val JArray(txs) = json \ "tx"
+            txs(txIndex).extract[String]
+          } getOrElse ByteVector32.Zeroes.toHex
         }
-      }
-    } yield ValidateResult(c, Right((Transaction.read(tx), fundingTxStatus)))
+        _ = span1.finish()
+        span2 = Kamon.spanBuilder("getrawtx").start()
```

Comment:
It looks like we'd benefit a lot from a utility function in KamonExt to wrap a future in a span.
It would be nice to just write `tx <- KamonExt.wrapInSpan("getrawtx")(getRawTransaction(txid))` which would automatically do the `span.finish()`.
The name `wrapInSpan` isn't very good, needs some bikeshedding.